tuplex/codegen/src/IteratorContextProxy.cc

```diff
@@ -0,0 +1,494 @@
+                auto stepPtr = builder.CreateGEP(_env->getRangeObjectType(), iterableStruct, {_env->i32Const(0), _env->i32Const(2)});
+                auto step = builder.CreateLoad(stepPtr);
+                builder.CreateStore(builder.CreateSub(start, step), indexPtr);
+            } else {
+                builder.CreateStore(_env->i32Const(-1), indexPtr);
+            }
+
+            // store pointer to iterable struct
+            auto iterablePtr = builder.CreateGEP(iteratorContextType, iteratorContextStruct, {_env->i32Const(0), _env->i32Const(2)});
+            if(iterableType.isListType() || iterableType.isTupleType()) {
+                builder.CreateStore(iterable.val, iterableStruct);
+                builder.CreateStore(iterableStruct, iterablePtr);
+            } else {
+                builder.CreateStore(iterableStruct, iterablePtr);
+            }
+
+            // store length for string or tuple
+            if(iterableType == python::Type::STRING) {
+                auto iterableLengthPtr = builder.CreateGEP(iteratorContextType, iteratorContextStruct, {_env->i32Const(0), _env->i32Const(3)});
+                builder.CreateStore(builder.CreateSub(iterable.size, _env->i64Const(1)), iterableLengthPtr);
+            } else if(iterableType.isTupleType()) {
+                auto iterableLengthPtr = builder.CreateGEP(iteratorContextType, iteratorContextStruct, {_env->i32Const(0), _env->i32Const(3)});
+                builder.CreateStore(_env->i64Const(iterableType.parameters().size()), iterableLengthPtr);
+            }
+
+            auto* dl = new DataLayout(_env->getModule().get());
+            return SerializableValue(iteratorContextStruct, _env->i64Const(dl->getTypeAllocSize(iteratorContextType)));
+        }
+
+        SerializableValue IteratorContextProxy::initZipContext(LambdaFunctionBuilder &lfb, llvm::IRBuilder<> &builder,
+                                                           const std::vector<SerializableValue> &iterables,
+                                                           IteratorInfo *iteratorInfo) {
+            using namespace llvm;
+
+            if(iterables.empty()) {
+                // use dummy value for empty iterator
+                return SerializableValue(_env->i64Const(0), _env->i64Const(8));
+            }
+
+            auto iterablesType = iteratorInfo->argsType;
+            auto argsIteratorInfo = iteratorInfo->argsIteratorInfo;
+            llvm::Type *iteratorContextType = _env->getZipIteratorType(iterablesType, argsIteratorInfo);
+            if(iteratorContextType == _env->i64Type()) {
+                // empty iterator
+                return SerializableValue(_env->i64Const(0), _env->i64Const(8));
+            }
+            auto iteratorContextStruct = _env->CreateFirstBlockAlloca(builder, iteratorContextType, "zip_iterator_alloc");
+            // store pointers to iterator structs
+            for (int i = 0; i < iterablesType.parameters().size(); ++i) {
```

Comment:
use unsigned int or size_t, I think size() returns an unsigned type.

Reply:
Changed to size_t.